test/e2e/edit-app-settings.spec.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const path = require('path');
+const { expect } = require('chai');
+const request = require('request-promise-native');
+
+const { getProjectUrl } = require('./cht-docker-utils');
+const {
+  DEFAULT_PROJECT_NAME,
+  cleanupProject,
+  getProjectDirectory,
+  initProject,
+  runChtConf,
+} = require('./cht-conf-utils');
+
+describe('edit-app-settings', () => {
+  const projectName = DEFAULT_PROJECT_NAME;
+  const projectDirectory = getProjectDirectory(projectName);
+
+  before(async () => {
+    await initProject(projectName);
+  });
+
+  after(async () => {
+    await cleanupProject(projectName);
+  });
+
+  it('disables a language, recompile, and push app settings', async () => {
+    const url = await getProjectUrl(projectName);
+    const initialSettings = await request.get({ url: `${url}/api/v1/settings`, json: true });
+
+    // TODO: remove next line when we upgrade eslint and its `parserOptions.ecmaVersion` setting to parse syntax supported by node 18+
+    // eslint-disable-next-line no-undef
+    const baseSettings = structuredClone(initialSettings);
+    baseSettings.languages = baseSettings.languages.map(language => {
+      if (language.locale === 'en') {
+        language.enabled = false;
+      }
+
+      return language;
+    });
+    baseSettings.locale = 'fr';
+    baseSettings.locale_outgoing = 'fr';
+    await fs.promises.writeFile(
+      path.join(projectDirectory, 'app_settings/base_settings.json'),
+      JSON.stringify(baseSettings, null, 2),
+    );
```

Comment:
I think we are going to want a layer of util code for writing/reading these standard config files for a given `projectDirectory`.  Maybe these functions could go into `cht-conf-utils.js` or a different dedicated file so they can be re-used in other tests.